kong/tools/queue.lua

```diff
@@ -251,11 +251,21 @@ function Queue:process_once()
     end
   end
 
+  local batch = {unpack(self.entries, self.front, self.front + entry_count - 1)}
+  -- Guard against queue shrinkage during handler invocation by using math.min below.
```

Comment:
This is no longer needed.  The comment kind of pointed at the bug that you've discovered without me actually realizing that it was a bug.  With your change, we'll actually always be able to remove `entry_count` entries.